### PR TITLE
layers: Label VUID 02251

### PR DESCRIPTION
**Pull request by sfricke-samsung**
From spec

> if any call to vkGetPhysicalDeviceImageFormatProperties2 returns an error then imageCreateImageFormatPropertiesList is defined to be the empty list.

So error can be anything that isn't `VK_SUCCESS`

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -1607,14 +1607,14 @@ bool CoreChecks::PreCallValidateCreateImage(VkDevice device, const VkImageCreate
         format_limits = image_format_properties.imageFormatProperties;
     }
 
-    if (result == VK_ERROR_FORMAT_NOT_SUPPORTED) {
+    if (result != VK_SUCCESS) {
 #ifdef VK_USE_PLATFORM_ANDROID_KHR
         if (!lvl_find_in_chain<VkExternalFormatANDROID>(pCreateInfo->pNext))
 #endif  // VK_USE_PLATFORM_ANDROID_KHR
-            skip |= LogError(device, kVUID_Core_Image_FormatNotSupported,
+            skip |= LogError(device, "VUID-VkImageCreateInfo-imageCreateMaxMipLevels-02251",
```
Comment (Tony-LunarG) [inline layers/buffer_validation.cpp:1614]: 02251 says nothing about unsupported format, and instead talks about max MipLevels, ArrayLayers, Extent and SampleCount.  Are you sure 02251 is better than Core_Image_FormatNotSupported?

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:1614]: So there is some spec gymnastics here so let me know if my logic sounds right on this (sorry, probably should have had this in the initial PR comment/commit message):

- 02251 states those values "must not be undefined" according to the image creation limit section above

- All of the 4 items listed have in the image creation limit
> The value is undefined if imageCreateImageFormatPropertiesList is empty.

- Above it states
> If any call to vkGetPhysicalDeviceImageFormatProperties2 returns an error, then imageCreateImageFormatPropertiesList is defined to be the empty list.

- `vkGetPhysicalDeviceImageFormatProperties2` only success code is `VK_SUCCESS`

So I was going to implement this and realize that `Core_Image_FormatNotSupported` is doing the same thing

Comment (Tony-LunarG) [reply layers/buffer_validation.cpp:1614]: Ok, with a couple of backflips I see it now.  Could you please add a comment to the code?  Also, do you know if the return value from  vkGetPhysicalDeviceImageFormatProperties2 is the only reason that the values in 02251 would be undefined or is there more 02251 to do elsewhere?

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:1614]: I will for sure add the comments. I will also do another review of that VUID as well, I at one point did a bit of history reading in the internal issue/PR that added it and that should have all the parts to this VU

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:1651]: Added comment, did a double check of Spec and history of addition. This was added to help handle the VUs `VK_EXT_image_drm_format_modifier` added, but also got tacked on when the `VK_ANDROID_external_memory_android_hardware_buffer` extension was added.

I get the sense this VUID was added as a way to cover the case where `vkGetPhysicalDeviceImageFormatProperties[2]` fails, but there is some spec-sugar language to handle the fun case of external formats when using AHB's

**Pull request merged**